jlm/llvm/ir/operators/lambda.hpp

```diff
@@ -218,43 +215,80 @@ class node final : public rvsdg::StructuralNode
     return GetOperation().attributes();
   }
 
-  [[nodiscard]] size_t
-  ncvarguments() const noexcept
-  {
-    return ninputs();
-  }
+  /**
+   * \brief Adds a context/free variable to the lambda node.
+   *
+   * \param origin
+   *   The value to be bound into the lambda node.
+   *
+   * \pre
+   *   \p origin must be from the same region as the lambda node.
+   *
+   * \return The context variable argument of the lambda abstraction.
+   */
+  ContextVar
+  AddContextVar(jlm::rvsdg::output * origin);
 
-  [[nodiscard]] size_t
-  nfctarguments() const noexcept
-  {
-    return subregion()->narguments() - ninputs();
-  }
+  /**
+   * \brief Maps input to context variable.
+   *
+   * \param input
+   *   Input to the lambda node.
+   *
+   * \returns
+   *   The context variable description corresponding to the input.
+   *
+   * \pre
+   *   \p input must be input to this node.
+   *
+   * Returns the context variable description corresponding
+   * to this input of the lambda node. All inputs to the lambda
+   * node are by definition bound context variables that are
+   * accessible in the subregion through the corresponding
+   * argument.
+   */
+  [[nodiscard]] ContextVar
+  MapInputContextVar(const rvsdg::input & input) const noexcept;
 
-  [[nodiscard]] size_t
-  nfctresults() const noexcept
-  {
-    return subregion()->nresults();
-  }
+  /**
+   * \brief Maps bound variable reference to context variable
+   *
+   * \param output
+   *   Region argument to lambda subregion
+   *
+   * \returns
+   *   The context variable description corresponding to the argument
+   *
+   * \pre
+   *   \p output must be an argument to the subregion of this node
+   *
+   * Returns the context variable description corresponding
+   * to this bound variable reference in the lambda node region.
+   * Note that some arguments of the region are formal call arguments
+   * and do not have an associated context variable description.
+   */
+  [[nodiscard]] std::optional<ContextVar>
+  MapBinderContextVar(const rvsdg::output & output) const noexcept;
```

Comment:
The same here, a comment would be nice that states that the owner of the output is the lambda subregion.

Reply:
Same, there is already a "\pre" doxygen section.